app/controllers/publishers_controller.rb

```diff
@@ -318,12 +344,28 @@ def show
 
   # Domain verified. See balance and submit payment info.
   def home
-    if current_publisher.promo_stats_status == :update
-      SyncPublisherPromoStatsJob.perform_later(publisher: current_publisher)
-    end
+    SyncPublisherPromoStatsJob.perform_later(publisher: current_publisher) if current_publisher.promo_stats_status == :update
+
     # ensure the wallet has been fetched, which will check if Uphold needs to be re-authorized
     # ToDo: rework this process?
-    current_publisher.wallet
+
+    wallet = current_publisher.wallet.wallet_json
+
+    # Create Uphold cards if they need to be created
+    if current_publisher.uphold_verified && current_publisher.default_currency_confirmed_at.present?
+      if should_create_default_currency_card?
+        UpholdServices::CardCreationService.new(publisher: current_publisher,
+                                                currency_code: current_publisher.default_currency).perform
+      end
+
+      if should_create_bat_card?
```

Comment:
As discussed in a DM we might want to move the flow logic from `should_create_bat_card?` to this method in the case of a BAT card not being the default.